File: dotnet/src/Pactman.NonprofitCheckPlus/Models/SingleCheckResult.cs
using System;
using System.Collections.Generic;

using Pactman.NonprofitCheckPlus.Internal;

namespace Pactman.NonprofitCheckPlus.Models
{
    /// <summary>
    /// The result of <see cref="NonprofitsResource.CheckAsync"/>.
    /// </summary>
    public sealed class SingleCheckResult : PactmanResult
    {
        /// <summary>Initializes the result.</summary>
        /// <param name="checkCount">Checks consumed so far in the current billing cycle.</param>
        /// <param name="timeTakenMs">Server-side processing time in milliseconds, when reported.</param>
        /// <param name="errors">Item-level failures reported alongside the response.</param>
        /// <param name="requestId">Correlation identifier from the response headers, when present.</param>
        /// <param name="status">HTTP status of the response.</param>
        /// <param name="raw">The unmodified parsed response body.</param>
        /// <param name="nonprofit">The organization, or <see langword="null"/> when the API returned no record.</param>
        public SingleCheckResult(
            int? checkCount,
            double? timeTakenMs,
            IReadOnlyList<ApiErrorDetail> errors,
            string? requestId,
            int status,
            ResponseBody raw,
            Nonprofit? nonprofit)
            : base(checkCount, timeTakenMs, errors, requestId, status, raw)
        {
            Nonprofit = nonprofit;
        }

        /// <summary>The organization, or <see langword="null"/> when the API returned no record.</summary>
        public Nonprofit? Nonprofit { get; }

        /// <inheritdoc />
        public override IReadOnlyDictionary<string, object?> ToDictionary() =>
            new FieldMap(base.ToDictionary())
            {
                ["nonprofit"] = Nonprofit?.ToDictionary(),
            };
    }
}

File: java/sdk/src/main/java/org/pactman/nonprofitcheckplus/models/SingleCheckResult.java
package org.pactman.nonprofitcheckplus.models;

import java.util.List;

/**
 * The result of
 * {@link org.pactman.nonprofitcheckplus.NonprofitsResource#check(String)}.
 */
public final class SingleCheckResult extends PactmanResult {

    private final Nonprofit nonprofit;

    /**
     * Initializes the result.
     *
     * @param nonprofit   the organization, or {@code null} when the API returned no record.
     * @param checkCount  checks consumed in the current billing cycle, or {@code null}.
     * @param timeTakenMs server-side processing time, or {@code null}.
     * @param errors      item-level failures the API reported alongside the response.
     * @param requestId   correlation identifier from the response headers, or {@code null}.
     * @param status      the HTTP status of the response.
     * @param raw         the unmodified parsed response body.
     */
    public SingleCheckResult(
            Nonprofit nonprofit,
            Integer checkCount,
            Double timeTakenMs,
            List<ApiErrorDetail> errors,
            String requestId,
            int status,
            ResponseBody raw) {
        super(checkCount, timeTakenMs, errors, requestId, status, raw);
        this.nonprofit = nonprofit;
    }

    /**
     * The organization the API matched.
     *
     * @return the organization, or {@code null} when the API returned no record.
     */
    public Nonprofit getNonprofit() {
        return nonprofit;
    }
}
